File: 1_data_manipulation/dplyr/02_read_from_table.R
# 

# # Read data from a Hive table into an R data frame

# ## Using odbc and DBI

# Load packages
library(odbc)
library(DBI)

# Connect to Impala with ODBC
impala <- dbConnect(
  drv = odbc(),
  dsn = "Impala DSN"
)

# Read data into a data frame
games <- dbGetQuery(impala, "SELECT * FROM games")

# Disconnect
dbDisconnect(impala)

# View the data as a data frame
games

# View the data as a tibble
library(tibble)
as.tibble(games)

File: 1_data_manipulation/pandas/02_read_from_table.py
# 

# # Read data from a Hive table into a pandas DataFrame

# Import modules
import numpy as np
import pandas as pd


# ## Using impyla

import impala.dbapi
con = impala.dbapi.connect(host='worker-1', port=21050)
sql = 'SELECT * FROM games'
games = pd.read_sql(sql, con)
games


# ## Using pyodbc

import pyodbc
con = pyodbc.connect('DSN=Impala DSN', autocommit=True)
sql = 'SELECT * FROM games'
games = pd.read_sql(sql, con)
games
